File: studycollab-app/app/src/main/java/com/example/studycollab/data/model/Course.kt
package com.example.studycollab.data.model

import com.google.gson.annotations.SerializedName

data class Course(
    @SerializedName("_id")
    val id: String,
    val name: String,
    val code: String,
    val semester: String,
    val lecturerName: String? = null,
    val totalLectures: Int = 13,
    val completedLectures: Int = 0,

    // שומרים על רשימת הנושאים שהיתה לך
    val topics: List<Topic> = emptyList(),

    // --- החדש: מערכת שעות ---
    val schedule: Schedule? = null
)

// המחלקה החדשה שמייצגת את הזמנים (יום, שעה, מיקום)
data class Schedule(
    val day: String,       // e.g., "Sunday"
    val startTime: String, // e.g., "08:00"
    val endTime: String,   // e.g., "11:00"
    val location: String   // e.g., "Building 3, Room 101"
)

// המחלקה הקיימת (לא נגענו בה)
data class Topic(
    val title: String,
    val isCompleted: Boolean
)

File: backend/src/models/Course.js
const mongoose = require('mongoose');

const TopicSchema = new mongoose.Schema({
    title: { type: String, required: true },
    isCompleted: { type: Boolean, default: false }
});

const CourseSchema = new mongoose.Schema({
    name: { type: String, required: true },
    code: { type: String, required: true },
    semester: { type: String, required: true },
    
    // --- התיקון: הגדרה רשמית של השדה lecturer ---
    lecturer: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'User',
        required: true 
    },
    // ---------------------------------------------

    schedule: {
        day: { type: String, required: true },
        startTime: { type: String, required: true },
        endTime: { type: String, required: true },
        location: { type: String, default: "TBD" }
    },
    
    totalLectures: { type: Number, default: 13 },
    completedLectures: { type: Number, default: 0 },
    topics: [TopicSchema],
    
    materials: [{
        title: String,
        url: String,
        type: { type: String, enum: ['link', 'file', 'zoom'] },
        uploadedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User' }
    }]
});

CourseSchema.index({ code: 1, semester: 1 }, { unique: true }); 

module.exports = mongoose.model("Course", CourseSchema);